x-pack/legacy/plugins/index_management/public/components/mappings_editor/components/document_fields/fields/field_types/boolean_type.tsx

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React from 'react';
+
+import { i18n } from '@kbn/i18n';
+
+import { NormalizedField, Field as FieldType } from '../../../../types';
+import { getFieldConfig } from '../../../../lib';
+import {
+  StoreParameter,
+  IndexParameter,
+  DocValuesParameter,
+  BoostParameter,
+} from '../../field_parameters';
+import { EditFieldSection, EditFieldFormRow, AdvancedSettingsWrapper } from '../edit_field';
+
+const getDefaultValueToggle = (param: string, field: FieldType) => {
+  switch (param) {
+    case 'boost': {
+      return field[param] !== undefined && field[param] !== getFieldConfig(param).defaultValue;
+    }
+    case 'null_value': {
+      return field.null_value !== undefined && field.null_value !== '';
+    }
+    default:
+      return false;
+  }
+};
+
+interface Props {
+  field: NormalizedField;
+}
+
+export const BooleanType = ({ field }: Props) => {
+  return (
+    <>
+      <EditFieldSection>
+        <StoreParameter />
+        <IndexParameter hasIndexOptions={false} />
+        <DocValuesParameter />
+      </EditFieldSection>
+
+      <AdvancedSettingsWrapper>
+        <EditFieldSection>
+          {/* boost */}
+          <BoostParameter defaultToggleValue={getDefaultValueToggle('boost', field.source)} />
+
+          {/* null_value */}
+          <EditFieldFormRow
```

Comment:
There is an existing component for `null_value`. You can replace this block by

```js
<NullValueParameter
  defaultToggleValue={getDefaultValueToggle('null_value', field.source)}
/>
```

Aside, there is no input to allow the user to set the null value 😊

Reply:
`null_value` is different depending on the datatype. if i'm understanding the documentation correctly, for boolean, it can only be `true` or `false`. 

https://www.elastic.co/guide/en/elasticsearch/reference/current/boolean.html

Reply:
Maybe it makes more sense to have a select field here with `true` and `false` as options?

Reply:
I see, we would probably then have to upgrade the `<NullValueParameter  />` component to allow a child input to be customized. But we need to be able to `Set the null value` as the toggle label indicates and be consistent with other places.

I would see something like this (not tested)

```js
<NullValueParameter
  defaultToggleValue={getDefaultValueToggle('null_value', field.source)}
>
  <UseField path="null_value" component="TextField" config={configWithCorrectValifation} />
</NullValueParameter>
```

Maybe a select with `true`, `"true"`, `false`, `"false"` which are the 4 allowed values. WDYT?

Reply:
👍 I think that makes sense. I'll work on this.